### PR TITLE
Clean up 32-bit ass-u-mptions for x64 and arm64.

### PR DESCRIPTION
**Pull request by barnson**

### CODE REVIEW

File: src/dutil/inc/strutil.h [hunk 1]
```diff
@@ -46,29 +46,29 @@ HRESULT DAPI StrAnsiTrimWhitespace(
 HRESULT DAPI StrAllocString(
     __deref_out_ecount_z(cchSource+1) LPWSTR* ppwz,
     __in_z LPCWSTR wzSource,
-    __in DWORD_PTR cchSource
+    __in SIZE_T cchSource
```
Comment (rseanhall) [inline src/dutil/inc/strutil.h:49]: I copied the usage of `DWORD_PTR` for string lengths in `verutil`.

Comment (barnson) [reply src/dutil/inc/strutil.h:49]: Hmm...I waffled but settled on `SIZE_T` being more correct. @robmen, opinion? Likely both work as ARM64 Burn works with verutil as it is today.

Comment (rseanhall) [reply src/dutil/inc/strutil.h:49]: Sorry, I wasn't trying to say `DWORD_PTR` was better. Just trying to point out more places that need to be updated.

Comment (barnson) [reply src/dutil/inc/strutil.h:49]: Got it. Though it's harder than it sounds, as a lot of errors won't kick off until you actually use the code from 64-bit and `DWORD_PTR` and `SIZE_T` are compatible...Well, searching for `cch` might be mostly sufficient.

Comment (robmen) [reply src/dutil/inc/strutil.h:49]: I seem to remember `SIZE_T` forcing callers to change their types to `SIZE_T` to compile and that was disruptive. I'll go with whichever way you guys think best.

**Pull request merged**